src/menu/src/interface.ts

```diff
@@ -9,13 +9,14 @@ export interface MenuOptionBase {
   icon?: () => VNodeChild
   children?: Array<MenuOption | MenuGroupOption>
   extra?: string | (() => VNodeChild)
+  type?: 'group' | 'submenu' | undefined
   [key: string]: unknown
   /** @deprecated */
   titleExtra?: string | (() => VNodeChild)
 }
 
 export interface MenuGroupOptionBase extends MenuOptionBase {
-  type: 'group'
+  type: 'group' | 'submenu'
   children: Array<MenuOption | MenuGroupOption>
 }
```

Comment:
We'd better not mix 'group' and 'submenu' together in one type.

Reply:
Well, yes, but in this case, there is a problem

I cannot extend DropdownGroupOption from MenuGroupOptionBase if it does not include this.

Should I make another type like the following?

```ts
// menu/src/interface.ts
export interface MenuGroupOptionBaseAbstract extends MenuOptionBase {
  type: 'group' | 'submenu'
  children: Array<MenuOption | MenuGroupOption>
}
export interface MenuGroupOptionBase extends MenuGroupOptionBaseAbstract {
  type: 'group'
}
// dropdown/src/interface.ts
export interface DropdownGroupOption extends MenuGroupOptionBaseAbstract {
  type: 'submenu'
}
```

Reply:
`type=submenu` is actually not required for `menu`, it's a compitable workaround.

Remove it seems to be okay.

Reply:
Yes, it's ok for menu, but then dropdown cannot extend it. I just tried it and typescript show this error when building the package:

```
src/dropdown/src/interface.ts:12:18 - error TS2430: Interface 'DropdownGroupOption' incorrectly extends interface 'MenuGroupOptionBase'.
  Types of property 'type' are incompatible.
    Type '"submenu"' is not assignable to type '"group"'.

12 export interface DropdownGroupOption extends MenuGroupOptionBase {
                    ~~~~~~~~~~~~~~~~~~~
Found 1 error.
```

Reply:
What if just remove 'submenu' from types?

Reply:
@07akioni Alright, done! 👍